Source/Scene/ModelExperimental/GeometryPipelineStage.js

```diff
@@ -446,4 +593,182 @@ function handleBitangents(shaderBuilder, attributes) {
   );
 }
 
+const scratchProjectedPosition = new Cartesian3();
+const scratchCartographic = new Cartographic();
+
+function projectPositionTo2D(position, modelMatrix, projection, result) {
```

Comment:
Hm, this file is getting rather large. Though I do see it's a bit tricky to separate  the 2D details from the basic geometry details.

Though can any of these helper functions be moved to their own utility module for map projections?

Reply:
I think I can repurpose `SceneTransforms.computeActualWgs84Position` here; it's basically the same thing, though I would have to force the `frameState.mode` to be `SceneMode.COLUMBUS_VIEW` so that the function doesn't drop the x-coordinate.

Reply:
I also wonder if it's possible to create the 2D buffer somewhere else, then just reference it in the `GeometryPipelineStage`. I'm not sure where that would go, though.